visit_attribute_type.md

```diff
@@ -0,0 +1,126 @@
+
+    Parameter | Type | Description
+    --- | --- | ---
+    *name* | `String` | Name of the visit attribute type (Required)
+    *description* | `String` | Description (Required)
+    *datatypeClassname* | `CustomDataType Resource` | Data type Classname of custom data type resource (Required)
+    *minOccurs* | `Number` | No of minimum occurrence (Required)
+    *maxOccurs* | `Number` | No of maximum occurrence
+    *preferredHandlerClassname* | `Handler` | Handler sub resource of CustomDataType   
+    *datatypeConfig* | `String` | Data type configuration   
+    *handlerConfig* | `String` | Handler config for intiate
+   
+    ```console
+        POST /visitattributetype
+       {
+         "name": "string",
+         "description": "string",
+         "datatypeClassname": "string",
+         "minOccurs": 0,
+         "maxOccurs": 0,
+         "datatypeConfig": "string",
+         "preferredHandlerClassname": "string",
+         "handlerConfig": "string"
+       }
+    ```
+### Update a visit attribute type
+
+*  Update a target visit attribute type with given UUID, this method only modifies properties in the request. Returns a `404 Not Found` 
+status if visit attribute not exists. If user not logged in to perform this action, a `401 Unauthorized` status returned.
+
+    #### Query Parameters
+
+    Parameter | Type | Description
+    --- | --- | ---
+    *uuid* | `target_visit_attribute_type_uuid` | Target visit attribute type resource UUID
+    
+    #### Attributes
+
+      Parameter | Type | Description
+      --- | --- | ---
+      *name* | `String` | Name of the visit attribute type (Required)
+      *description* | `String` | Description (Required)
+      *datatypeClassname* | `CustomDataType Resource` | Data type Classname of custom data type resource (Required)
+      *minOccurs* | `Number` | No of minimum occurrence (Required)
+      *maxOccurs* | `Number` | No of maximum occurrence
+      *preferredHandlerClassname* | `Handler` | Handler sub resource of CustomDataType   
+      *datatypeConfig* | `String` | Data type configuration   
+      *handlerConfig* | `String` | Handler config for intiate
+    
+    ```console
+        POST /visitattributetype/:target_visit_attribute_type_uuid
+        {
+          "name": "string",
+          "description": "string",
+          "datatypeClassname": "string",
+          "minOccurs": 0,
+          "maxOccurs": 0,
+          "datatypeConfig": "string",
+          "preferredHandlerClassname": "string",
+          "handlerConfig": "string"
+        }
+    ```
+    
+### Delete a visit attribute type
+
+* Delete or Retire a target visit attribute type by its UUID. Returns a `404 Not Found` status if visit attribute not exists.If user not logged 
+  in to perform this action, a `401 Unauthorized` status returned.
+
+    #### Query Parameters
+
+    Parameter | Type | Description
+    --- | --- | ---
+    *purge* | `Boolean` | The resource will be voided/retired unless purge = ‘true’
```

Comment:
Purging will attempt to irreversibly remove the attribute type from the system. Attribute types that have been used (i.e., are referenced from existing data) cannot be purged.